app/views/support/WeeklyPromotion.scala

```diff
@@ -41,38 +46,66 @@ object WeeklyPromotion {
       )
     }
-    val UKandUS = Seq(DiscountedRegion(
-      title = "United Kingdom",
-      description = "Includes Isle of Man and Channel Islands",
-      countries = UK,
-      discountedPlans = plansForPromotion(promotion, promoCode, UK, catalog.weeklyZoneA.toList)
-    ), DiscountedRegion(
+    val UKregion: Set[DiscountedRegion] = {
+      val all = DiscountedRegion(
+        title = "United Kingdom",
+        description = "Includes Isle of Man and Channel Islands",
+        countries = UK,
+        discountedPlans = plansForPromotion(promotion, promoCode, UK, catalog.weeklyZoneA.toList)
+      )
+      val domestic = DiscountedRegion(
+        title = "United Kingdom",
+        description = "Includes mainland UK only.",
+        countries = UKdomestic,
+        discountedPlans = plansForPromotion(promotion, promoCode, UK, catalog.weeklyZoneA.toList)
+      )
+      val overseas = DiscountedRegion(
+        title = "Isle of Man and Channel Islands",
+        description = "",
+        countries = UKoverseas,
+        discountedPlans = plansForPromotion(promotion, promoCode, UK, catalog.weeklyZoneA.toList)
+      )
+      val countries = promotionCountries intersect UK
+      val isDomestic = !(countries intersect UKdomestic).isEmpty
+      val isOverseas = !(countries intersect UKoverseas).isEmpty
+      if(isDomestic && isOverseas){
+        Set(all)
+      } else if (isDomestic) {
+        Set(domestic)
+      } else if (isOverseas) {
+        Set(overseas)
+      } else {
+        Set()
+      }
+    }
+    val USregion = DiscountedRegion(
       title = "United States",
       description = "Includes Alaska and Hawaii",
       countries = US,
       discountedPlans = plansForPromotion(promotion, promoCode, UK, catalog.weeklyZoneA.toList)
-    ))
-    val regions = regionForZoneCCountry ++ UKandUS ++ Seq(rowWithoutCountry)
+    )
+    val regions = regionForZoneCCountry ++ UKregion ++ Seq(USregion) ++ Seq(rowWithoutCountry)
     regions.filter(_.discountedPlans.length > 0)
   }
 
   private val currencyFor = CountryGroup.availableCurrency(Currency.all.toSet) _
 
-  def plansForPromotion(promotion: PromoWithWeeklyLandingPage, promoCode: PromoCode, countries: Set[Country], catalogPlans: Seq[CatalogPlan.Paid])(implicit catalog: Catalog): List[DiscountedPlan] = {
-    def currencies = (countries intersect promotion.appliesTo.countries).flatMap { country => currencyFor(country) }
+  def plansForPromotion(promotion: Option[PromoWithWeeklyLandingPage], promoCode: PromoCode, countries: Set[Country], catalogPlans: Seq[CatalogPlan.Paid])(implicit catalog: Catalog): List[DiscountedPlan] = {
+    def currencies = promotion.map(countries intersect _.appliesTo.countries).getOrElse(countries).flatMap { country => currencyFor(country) }
+    val productRatePlanIds =(productRatePlanId: ProductRatePlanId) => promotion.map(_.appliesTo.productRatePlanIds.contains(productRatePlanId)).getOrElse(true)
```

Comment:
if you make this a def that takes a parameter, the the syntax will be slightly neater